src/neural/network_opencl.cc

```diff
@@ -134,111 +134,102 @@ class OpenCLNetwork : public Network {
     params_.tune_exhaustive =
         options.GetOrDefault<bool>("tune_exhaustive", false);
 
-    const int inputChannels = kInputPlanes;
     const int channels = weights.input.biases.size();
-    const size_t residual_blocks = weights.residual.size();
 
-    /*
```

Comment:
This comment block was meant to remind the people working of this backend, the usual values they were dealt with.
```
/*
     static constexpr int NUM_VALUE_INPUT_PLANES = 32;
     static constexpr int NUM_POLICY_INPUT_PLANES = 32;
     static constexpr int NUM_OUTPUT_POLICY = 1858;
     static constexpr int NUM_VALUE_CHANNELS = 128;
*/
```

Reply:
What do the "usual" values matter? What matters from a code perspective is that the size of the blocks/data must be correct, and if we can read those sizes from the `Weights` struct, much better to do that than to think about the actual underlying numbers. Especially since those numbers are subject to change, and in fact *did* change in the v1 -> v2 transition. Reliance on the global constants is the only reason that horrific "input planes bug" was even possible in the first place. Using dynamic sizes from the `Weights` struct makes that class of bug entirely impossible, and adding back the actual numbers, even in a comment, serves no purpose.

Reply:
Actually, that 1858 is used over all places in the code.
I think since long ago that it should be replaced with a constant in network.h, but didn't have good enough occasion to do that. So if anyone will edit somewhere there, that would be good thing to change too.

Reply:
Perhaps it's used in other places, but it's not used in network_{opencl,blas}.cc, and has no reason to be when the *actual* weights size is available. Agreed it's good to consolidate the other uses where the actual weights aren't available.